src/api.jl

```diff
@@ -0,0 +1,10962 @@
+"""
```

Comment:
this file is not auto generated?

Reply:
I generated it using a script initially, and have been adding stuff to it manually now. If I need to edit all the functions I either edit it with another script or just using editor magic. This can probably be done better, but at the moment I don't know what patterns exist and what can be abstracted so I'm just going down the dumb and verbose path.

Reply:
I see, since we want to be extensive, we might consider automating that in the future. Ok lets keep it that way for now

Reply:
After exploring Clang.jl source and documentation more, I believe we can autogenerate `api.jl` (and least most of it) directly from Clang.jl. I'll keep proceeding with the current implementation for now though. I will submit a separate PR at a later stage where all this is auto generated.